sim_plots.py

```diff
@@ -98,8 +82,33 @@
         theme="streamlit",
     )
 
+
+while True:
+    try:
+        sim_plotter.load_state()
+        break
+    except Exception as e:
+        time.sleep(3)
+        title.title(f"Waiting for sim state... {e}")
+        continue
+
+while True:
+    try:
+        st, new_ts = sim_plotter.load_state()
+    except EOFError:
+        time.sleep(1)
+        continue
+
+    if new_ts == last_ts:
+        time.sleep(1)
+        continue
+
+    load_canvas_on_state(new_ts)
     last_ts = new_ts
 
     if last_ts == "final":
-        title.title("Final sim state")
+        snapshots = SimPlotter.available_snapshots()
+        timestamp = inputs.select_slider("Go to snapshot", snapshots, value="final")
+        st, new_ts = sim_plotter.load_state(timestamp)
+        load_canvas_on_state(timestamp)
```

Comment:
i might not have waited until the very end to see this come up.
This looks good, but i did not verify/test the slider.